### PR TITLE
Feature/77 fft

### LINKED ISSUE #77: Add function(s) for computing a FFT
Suggest a function using scipy's welch which allows specifying the bin width, but won't return phase angle, and another function using Numpy's rfft that would return phase.

Here are two examples:

~~~python
def welch_fft(df, bin_width=0.5):
  fft = endaq.calc.psd.welch(df, bin_width=bin_width, scaling='parseval', window='boxcar', noverlap=0)
  return fft**0.5 * (2**0.5) #scale from g^2 as RMS to g-peak  

def numpy_fft(df):
  """
  Using Numpy's rfft functions compute a discrete Fourier Transform
  """
  freq = np.fft.rfftfreq(len(df), d=endaq.calc.utils.sample_spacing(df))
  df_mag = pd.DataFrame(index=pd.Series(freq, name="frequency (Hz)"), columns=df.columns)
  df_phase = df_mag.copy()

  for col in df.columns:
    sp = np.fft.rfft(df[col].to_numpy())
    df_mag[col] = np.absolute(sp)/len(sp)
    df_phase[col] = np.angle(sp)

  return df_mag, df_phase
~~~

### PR DESCRIPTION
**Pull request by CFlaniganMide**
This PR adds functionality to `endaq.calc` for FFT's, and possibly some other functions in the `scipy.fft` module such as the DCT.

Closes #77

### CODE REVIEW

**Review by shanlyMIDE — APPROVED** [review]
looks good, let's get it in - I want to promote it!

File: endaq/calc/fft.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+def aggregate_fft(df, **kwargs):
+    """
+    Calculate the FFT using `scipy.signal.welch` with a specified frequency spacing.  The data returned is in the same
+    units as the data input.
+
+    :param df: the input data
+    :param bin_width: the desired width of the resulting frequency bins, in Hz;
+        defaults to 1 Hz
+    :param kwargs: other parameters to pass directly to ``scipy.signal.welch``
+    :return: a periodogram
```
Comment (StokesMIDE) [inline endaq/calc/fft.py:25]: It might be good to describe the data a little more

File: endaq/calc/fft.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+    """
+    Perform the FFT of the data in `df`, using Scipy's FFT method from `scipy.fft.fft`.  If the in `df` is all real,
+    then the output will be symmetrical between positive and negative frequencies, and it is instead recommended that
+    you use the `rfft` method.
+
+    :param df: the input data
+    :param output: *Optional*  The type of the output of the FFT. Default is "magnitude".  "magnitude" will return the
+                   absolute value of the FFT, "angle" will return the phase angle in radians, "complex" will return the
+                   complex values of the FFT.
+    :param nfft: *Optional* Length of the transformed axis of the output. If nfft is smaller than the length of the
```
Comment (StokesMIDE) [inline endaq/calc/fft.py:54]: The `*Optional*` note (here and elsewhere) seems redundant; all keyword arguments are inherently optional. The type hinting also notes it.

**Review by StokesMIDE — COMMENTED** [review]
Minor comments made on docstrongs. Additionally, for consistency, the parameter descriptions should probably start with a capital letter, like sentences.

**Review by StokesMIDE — APPROVED** [review]

**Pull request merged**